eo-maven-plugin/src/main/java/org/eolang/maven/ParseMojo.java

```diff
@@ -149,46 +150,51 @@ private void parse(final ForeignTojo tojo) throws IOException {
-                    new Syntax(
-                        name,
-                        new InputOf(source),
-                        new OutputTo(baos)
-                    ).parse();
-                    final String parsed = new XMLDocument(
-                        new Xembler(
-                            new Directives().xpath("/program").attr(
-                                "source",
-                                source.toAbsolutePath()
-                            )
-                        ).applyQuietly(new XMLDocument(baos.toByteArray()).node())
-                    ).toString();
-                    Logger.debug(
-                        this,
-                        "Parsed program %s:\n %s",
-                        name,
-                        parsed
-                    );
-                    return parsed;
-                }
-            );
-        } catch (final ParsingException ex) {
+        footprint.save(
+            name,
+            "xmir",
+            () -> {
+                final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                new Syntax(
+                    name,
+                    new InputOf(source),
+                    new OutputTo(baos)
+                ).parse();
+                final String parsed = new XMLDocument(
+                    new Xembler(
+                        new Directives().xpath("/program").attr(
+                            "source",
+                            source.toAbsolutePath()
+                        )
+                    ).applyQuietly(new XMLDocument(baos.toByteArray()).node())
+                ).toString();
+                Logger.debug(
+                    this,
+                    "Parsed program %s:\n %s",
+                    name,
+                    parsed
+                );
+                return parsed;
+            }
+        );
+        final XML xmir = new XMLDocument(footprint.load(name, "xmir"));
+        final List<XML> errors = xmir.nodes("/program/errors/error");
+        if (!errors.isEmpty()) {
+            for (final XML error : errors) {
+                Logger.error(
+                    this,
+                    "Failed to parse '%s:%s': %s (just logging, because of failOnError=false)",
+                    source, error.xpath("@line").get(0), error.xpath("text()").get(0)
+                );
+            }
             if (this.failOnError) {
                 throw new IllegalArgumentException(
-                    String.format("Failed to parse %s", source),
-                    ex
+                    String.format(
+                        "Failed to parse %s (%d parsing errors)",
+                        source, errors.size()
+                    )
                 );
             }
-            Logger.warn(
-                this, "Parsing was skipped due to failOnError=false. In file %s with error: %s",
-                source,
-                ex.getMessage()
-            );
             return;
```

Comment:
@yegor256 Previously, we had to use the `return` statement because we didn't have any other option. However, now you can eliminate the problem of having two exit points by using the `if-else` statement.
```java
if (!errors.isEmpty()) {
...
} else {
...
}
```

Reply:
@volodya-lombrozo thanks, fixed: aebb9c2fa